docs/aws/audit/snsmonitoring/rules/is_topic_exposed_remediation.mdx

```diff
@@ -1,6 +1,227 @@
+
+2. **Select the SNS Topic:**
+   - In the SNS dashboard, click on "Topics" in the left-hand navigation pane.
+   - Select the SNS topic you want to configure from the list of topics.
+
+3. **Modify Access Policy:**
+   - In the topic details page, click on the "Access policy" tab.
+   - Review the existing access policy and ensure that it does not grant public access. Specifically, check for any `"Effect": "Allow"` statements that include `"Principal": "*"` which would allow access to everyone.
+
+4. **Restrict Access:**
+   - Edit the access policy to restrict access to only specific AWS accounts or IAM roles/users that require it. You can use the AWS Policy Generator to help create a more restrictive policy if needed.
+   - Save the changes to the access policy.
+
+By following these steps, you can ensure that your SNS topics are not exposed to unauthorized access.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent SNS Topics from being exposed in AWS using the AWS CLI, you can follow these steps:
+
+1. **List SNS Topics**:
+   First, identify the SNS topics in your account.
+   ```sh
+   aws sns list-topics
+   ```
+
+2. **Get Topic Attributes**:
+   For each topic, retrieve its attributes to check the current policy.
+   ```sh
+   aws sns get-topic-attributes --topic-arn <topic-arn>
+   ```
+
+3. **Update Topic Policy**:
+   Modify the topic policy to restrict access. Ensure that only specific AWS accounts or IAM roles/users have access. Here is an example of a restrictive policy:
+   ```sh
+   aws sns set-topic-attributes --topic-arn <topic-arn> --attribute-name Policy --attribute-value '{
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::<account-id>:root"
+         },
+         "Action": "SNS:Publish",
+         "Resource": "<topic-arn>"
+       }
+     ]
+   }'
+   ```
+
+4. **Verify Policy Update**:
+   Confirm that the policy has been updated correctly by retrieving the topic attributes again.
+   ```sh
+   aws sns get-topic-attributes --topic-arn <topic-arn>
+   ```
+
+By following these steps, you can ensure that your SNS topics are not exposed and are only accessible by authorized entities.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent SNS Topics from being exposed in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to ensure that your SNS topics are not exposed:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>